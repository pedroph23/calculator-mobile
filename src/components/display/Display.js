import React, { Component } from 'react';
import {View, Text, StyleSheet} from 'react-native';

export default class Display extends Component {
 
    render() {
    
        return(
            <View style= { Style.display  }> 
                <Text style={ Style.text } numberOfLines={ 1 }> 
                    { this.props.value }
                </Text >
            </View>
        );
    }
}

const Style = StyleSheet.create({
    text: {
        fontSize: 90,
        color: 'white',
    },
    display: {
        flex: 1,
        // padding: 20,
        justifyContent: "center",
        alignItems:"flex-end",
        borderColor: '#888',
        backgroundColor: '#464646',
        maxHeight: '40%'
    },
});