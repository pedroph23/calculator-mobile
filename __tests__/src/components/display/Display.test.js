import Display from '../../../../src/components/display/Display';
import {View} from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

beforeEach(() => {
    jest.resetModules();
    // console.log = () => {};
    // console.warn = () => {};
    // console.info = () => {};
    // console.error = () => {};
  });


  describe('Test class Display', () => {

    test('return elements in function render', () => {

      // Not tested 
      
        // Input
        const props ={
          value: '0'
        };

        // Prepare  and execute component
        const displayComponent = renderer.create(<Display />);
        const tree = displayComponent.toJSON();
        
        // Validate
        // expect(result).toEqual();
        expect(tree).toMatchSnapshot();
    });
  });
