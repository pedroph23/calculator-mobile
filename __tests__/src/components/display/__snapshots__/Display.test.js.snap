// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test class Display return elements in function render 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#464646",
      "borderColor": "#888",
      "flex": 1,
      "justifyContent": "center",
      "maxHeight": "40%",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "white",
        "fontSize": 90,
      }
    }
  />
</View>
`;
